mxcube3/ui/components/Tasks/Characterisation.js

```diff
@@ -1,137 +1,293 @@
+'use strict';
+
 import React from 'react'
 import {reduxForm} from 'redux-form';
-import { Modal } from "react-bootstrap";
+import Modal from 'react-modal';
 
 
 class Characterisation extends React.Component {
 
-    constructor(props) {
-        super(props);
-        this.state = {};
-        this.handleSubmit = this.handleSubmit.bind(this);
-    }
 
-    componentWillReceiveProps(nextProps) {
-        // this is to prevent infinite recursion, caused by calling initializeForm
-        // which makes componentWillReceiveProps to be called again
-        if (nextProps.clicked_task === this.state.clicked_task) { return }
-        
-        if (Object.keys(nextProps.clicked_task).length > 0) {
-            this.props.initializeForm(nextProps.clicked_task.parameters);
+  constructor(props) {
+        super(props);
+        this.runNow = this.handleSubmit.bind(this, true);
+        this.addToQueue = this.handleSubmit.bind(this, false);
+  }
+
+    handleSubmit(runNow){
+
+        let fields = this.props.fields;
+        let parameters = {
+            Type : "Characterisation",
+            numImages : fields.numImages.value,
+            expTime: fields.expTime.value,
+            resolution : fields.resolution.value,
+            oscStart : fields.oscStart.value,
+            energy : fields.energy.value,
+            oscRange : fields.oscRange.value,
+            transmission : fields.transmission.value,
+            point : this.props.pointId,
+            detectorMode: fields.detectorMode.value,
+            kappa: fields.kappa.value,
+            phi: fields.phi.value,
+            radiationDamage: fields.radiationDamage.value,
+            optSAD: fields.optSAD.value,
+            spaceGroup: fields.spaceGroup.value,
+            crystMin: fields.crystMin.value,
+            crystMax: fields.crystMax.value,
+            omegaMin: fields.omegaMin.value,
+            omegaMax: fields.omegaMax.value,
+            stratComp: fields.stratComp.value
+
+            };
+
```

Comment:
In fact, in my PR I did it like that:

`let parameters = Object.assign({Type: "Characterisation"}, this.props.values);`

`this.props.values` is something that is defined in redux form, it contains field names + values for each field (see documentation)

Did you notice a problem with it ? Maybe it's converting all values to string, but I am not sure.